### ISSUE TITLE
High difficulty on P2Pool node with Nvidia GPU

### ISSUE THREAD

**Issue by quinten-miller** [KlausT/ccminer#208]
I've noticed that the Share Difficulty on P2Pool nodes is significantly higher different users. Those sitting above the minimum difficulty are usually all sitting at the same number, regardless of hashrate or which pool / node they are connected. 

The only clue i have is someone mentioning that it is related to NVidia GPU's, which is what I'm running. 

I've had a poke around the P2Pool code (Python is not my language of choice :| ), however, i doubt the node knows what GPU it is connected... so I'm wondering if it is something that is being sent from the client? 

Is there a header or meta data sent to P2Pool and if so... could something in that be effecting the difficulty of mining node, such that the server ignores desired settings sent in the config and ignores the minimum of the node at that time. 

Thoughts appreciated

**Comment by KlausT** [comment]
As far as I know it works this way:
Only the number of shares per minute on the network affect the minimum share difficulty.
Fast miners should increase the share difficulty by adding /x to the username where x is the difficulty factor.
You can set the miner's pseudo share difficulty to the value x by adding +x to the username. This is not the same as /x
I don't think p2pool cares about any meta data from the miner.

**Comment by lyolyalya** [comment]
https://en.bitcoin.it/wiki/P2Pool
so p2pool is something like a big miner. its diff will be pretty high especially if there are no miners.

**Comment by quinten-miller** [comment]
Thanks for responding @KlausT 

I don't think this is a related to the network and hashrate of the miners. While they do have an impact on the difficulty, that is generally reflected in the minimum hashrate of the node. 

Below are three different P2Pool nodes, notice that each node has miners operating at the minimum hashrate and then miners operating higher. The miners operating higher than the minimum hashrate are all operating with approx 0.5 difficulty of each other, out of ~570. This is across mining node instances as well as the high hashrate node and the low hashrate node (as seen by the minimum difficulty). 

Miner A
![miner a](https://user-images.githubusercontent.com/767971/36399032-24c2afd2-161e-11e8-9f53-375135ded7fa.PNG)

Miner B
![miner b](https://user-images.githubusercontent.com/767971/36399033-24fb3528-161e-11e8-8b44-11a76cbb278c.PNG)

Miner C
![miner c](https://user-images.githubusercontent.com/767971/36399034-2532789e-161e-11e8-978b-658d02c108f9.PNG)

Basically, just start browsing the nodes list on https://scanner.vtconline.org/ and you will start to see the pattern. 

Such coincidences across such a variety of variable has made me think that this is an abnormal behavior. 

As mentioned, i'm looking at the P2Pool side of the equation, however I was wondering if you might have any insights as to why something like this may be occurring

**Comment by KlausT** [comment]
I have just tested it. Looks like a pool issue.
When I used only the address as username then there was a high share difficulty.
Then I have tried adding /1 to it. Suddenly the share difficulty was lower.

**Comment by quinten-miller** [comment]
Excellent!!!

Thanks heaps. I'll see about getting the docs updated.

**Comment by quinten-miller** [comment]
Actually, i was looking at the code last night that assigns the desired share difficulty, splitting on the '/'... maybe there is a bug that can be fixed :)
